gcc/rust/analysis/rust-scan.cc

```diff
@@ -353,7 +353,7 @@ TopLevelScan::visit (AST::UseDeclaration &use_decl)
 void
 TopLevelScan::visit (AST::Function &function)
 {
-  functions[function.function_name] = &function;
```

Comment:
thanks for doing this my branch to move to HIR is updating the pipeline with a new name/type resolution algorithm so we wont need those extra fields on LetStmt and Function inside the AST which i will remove with that PR